pkg/image/importer/credentials.go

```diff
@@ -183,6 +184,15 @@ func basicCredentialsFromKeyring(keyring credentialprovider.DockerKeyring, targe
 			glog.V(5).Infof("Being asked for %s, trying %s for legacy behavior", target, "docker.io")
 			return basicCredentialsFromKeyring(keyring, &url.URL{Host: "docker.io"})
 		}
+
+		// try removing the canonical ports
+		if strings.HasSuffix(target.Host, ":443") || strings.HasSuffix(target.Host, ":80") {
```

Comment:
Yeah, I think there are several cases:

schema = https
schema = http
schema = none
port = 443
port = 80
port = unspecified

For unspecified or canonical schema or port, the credentials should be sent to a secured server.      I think when schema is unspecified and the canonical port is used, we should infer the schema.  We should probably be careful not to send credentials to HTTP endpoints unless the user requests it, but on the other hand a user might reasonably expect a host like `foo.example.com` to be sent to the server.